modules/data.atmosphere/R/download.Geostreams.R

```diff
@@ -56,6 +56,25 @@ download.Geostreams <- function(outfolder, sitename,
   sensor_txt <- httr::content(sensor_result, as = "text", encoding = "UTF-8")
   sensor_info <- jsonlite::fromJSON(sensor_txt)
   sensor_id <- sensor_info$id
+
+  # Request that Geostreams recalculate data availability for us
+  # otherwise start and end dates may be null or stale
+  update_result <- httr::GET(url = paste0(url, "/sensors/", sensor_id, "/update"),
```

Comment:
can this be made optional, maybe with default being true. If I call this function a few times in a row I might want to update it the first time, but subsequent calls can just use the precomputed values.

Reply:
I went with the update-every-time theory after a discussion in the TERRA project Slack where @dlebauer posed that it's not PEcAn's job to know whether Clowder needs to recompute a value before sending it, and @max-zilla replied that `sensors/:id/update` is 'basically harmless'. If it's not harmless, we should revisit whether to call it at all. I see three possible alternatives:

* "Servers don't lie": Never call update, treat the reported availability dates as authoritative, and let users complain to Clowder admins if stale timestamps mean they can't see the data they want. 
* "Are you _sure_ you don't have that?": Compare user dates against the availability reported in the initial site lookup, and if user dates fall outside the reported range then call update and check again before giving up.
* "Servers don't lie but they do lag sometimes": Assume that data are added in ~real time and that the server updates at some regular interval, and call update only if `sensor_info$max_end_time < (Sys.time() - ASSUMED_UPDATE_INTERVAL)`. 

It doesn't change any of the above, but re "If I call this function a few times in a row": Note that for the use case of downloading multiple years from one site, the update already happens only once per `download.Geostreams` call and isn't repeated for each year.

Reply:
Most of the code that writes to the database will also update the sensor information, they know best when things up date. As @dlebauer said we can call this endpoint from a cronjob every so often if need to. In which case I'd go with option 1 "Servers don't lie".

As for the option it is not a no-op, however calling it for a single sensor does cut down the compute time, but the more datapoints there are the more expensive the function is. (it has to loop through all datapoints for a sensor, finding start time, end time and variables and combine all those values).